xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java

```diff
@@ -0,0 +1,91 @@
+    private static final String TENANT = "TEST";
+    private CommunicationMailService subject;
+
+    private TenantContextHolder tenantContextHolder;
+    private KafkaTemplateService kafkaTemplateService;
+
+    @Before
+    public void setup() {
+        tenantContextHolder = mock(TenantContextHolder.class);
+        kafkaTemplateService = mock(KafkaTemplateService.class);
+        mockTenant(TENANT);
+        subject = new CommunicationMailService(kafkaTemplateService, tenantContextHolder);
+        ReflectionTestUtils.setField(subject, "topicName", COMMUNICATION_KAFKA_TOPIC_PATTERN);
+    }
+
+    @Test
+    public void sendEmailEvent() {
+        String topicName = String.format(COMMUNICATION_KAFKA_TOPIC_PATTERN, TENANT);
+        CommunicationMessage communicationMessage = createCommunicationMessage(new ArrayList<>());
+        TemplateModel templateModel = createTemplateModel();
+        String convertedTemplateModel = convertToString(templateModel);
+        List<CommunicationRequestCharacteristic> characteristics = List.of(createCharacteristic("templateModel", convertedTemplateModel));
+        CommunicationMessage expectedMessage = createCommunicationMessage(characteristics);
+        String convertedExpectedMessage = convertToString(expectedMessage);
+
+        subject.sendEmailEvent(communicationMessage, templateModel);
+
+        verify(kafkaTemplateService).send(eq(topicName), eq(convertedExpectedMessage));
+        verifyNoMoreInteractions(kafkaTemplateService);
+    }
+
+    private void mockTenant(String tenant) {
+        TenantContext tenantContext = mock(TenantContext.class);
+        when(tenantContext.getTenantKey()).thenReturn(Optional.of(TenantKey.valueOf(tenant)));
+        when(tenantContextHolder.getContext()).thenReturn(tenantContext);
+        when(tenantContextHolder.getTenantKey()).thenReturn(tenant);
+    }
+
+    private CommunicationMessage createCommunicationMessage(List<CommunicationRequestCharacteristic> characteristics) {
+        CommunicationMessage communicationMessage = new CommunicationMessage();
+        communicationMessage.setSubject("Subject");
+        communicationMessage.setContent("content");
+        communicationMessage.setCharacteristic(characteristics);
+        return communicationMessage;
+    }
+
+    private TemplateModel createTemplateModel() {
+        TemplateModel templateModel = new TemplateModel();
+        templateModel.setTemplateKey("templateKey");
+        templateModel.setLangKey("en");
+        return templateModel;
+    }
+
+    private CommunicationRequestCharacteristic createCharacteristic(String name, String value) {
+        CommunicationRequestCharacteristic characteristic = new CommunicationRequestCharacteristic();
+        characteristic.setName(name);
+        characteristic.setValue(value);
+        return characteristic;
+    }
+
+    @SneakyThrows
+    private String convertToString(Object object) {
+        return new ObjectMapper().writeValueAsString(object);
```

Comment:
autowire objectMapper

Reply:
I autowired mapper in the CommunicationMailService but not in the tests. 
Does it required for the tests? In addition, I did not find configured object mapper in any configurations as well as in tests in xm-commons to be able to autowire it there + we use only one method to convert to string.